src/Framework/NativeMethods.cs

```diff
@@ -488,6 +514,22 @@ public SystemInformationData()
                                 {
                                     ProcessorArchitectureType = ProcessorArchitectures.ARM64;
                                 }
+                                else if (arch.StartsWith("s390x", StringComparison.OrdinalIgnoreCase))
```

Comment:
When is the OSArchitecture "unknown", but we can find arch via uname?

Reply:
Good question.

Anything that is excluded by these ifdefs:

```
#if !NET35
                // Get the architecture from the runtime.
                // Get the architecture from the runtime.
                processorArchitecture = RuntimeInformation.OSArchitecture switch
                processorArchitecture = RuntimeInformation.OSArchitecture switch
                {
                {
                    Architecture.Arm => ProcessorArchitectures.ARM,
                    Architecture.Arm => ProcessorArchitectures.ARM,
                    Architecture.Arm64 => ProcessorArchitectures.ARM64,
                    Architecture.Arm64 => ProcessorArchitectures.ARM64,
                    Architecture.X64 => ProcessorArchitectures.X64,
                    Architecture.X64 => ProcessorArchitectures.X64,
                    Architecture.X86 => ProcessorArchitectures.X86,
                    Architecture.X86 => ProcessorArchitectures.X86,
#if NET5_0_OR_GREATER
                    Architecture.Wasm => ProcessorArchitectures.WASM,
#endif
#if NET6_0_OR_GREATER
                    Architecture.S390x => ProcessorArchitectures.S390X,
#endif
#if NET7_0_OR_GREATER
                    Architecture.LoongArch64 => ProcessorArchitectures.LOONGARCH64,
                    Architecture.Armv6 => ProcessorArchitectures.ARMV6,
                    Architecture.Ppc64le => ProcessorArchitectures.PPC64LE,
#endif
                    _ => ProcessorArchitectures.Unknown,
                    _ => ProcessorArchitectures.Unknown,
                };
                };
#endif
#endif
```

Probably we can keep the `NETX_0_OR_GREATER` (leaving these unkown on older versions of .NET), and don't care about the `NET35` either?

Then we can remove all the `uname` cases that are already detected through `RuntimeInformation.OSArchitecture`.

Shall I make that change?

Reply:
@Forgind, shall I remove the known `uname` cases handled by `RuntimeInformation.OSArchitecture`?

Reply:
NET35 is fairly commonly used on windows...but it looks like all this code is explicitly non-windows. I'm guessing it's a remnant from mono. I'd keep the !NET35 check from earlier. If we end up falling back to Unknown, what actually goes wrong? If it isn't completely disastrous, I'd say yes to removing uname cases detected through RuntimeInformation.OSArchitecture, assuming that architecture is .NET Core 3.1.

Reply:
I think this is fine as-is. The net35 taskhost should not be used in any scenario targeting a non-.NET 3.5 project, so it can remain ignorant of these architectures.

Reply:
No, new MSBuild no longer runs on classic Mono (the Mono fork stopped updating a while ago), so we can cut that.